frontends/p4/toP4/toP4.cpp

```diff
@@ -1261,7 +1261,13 @@ bool ToP4::preorder(const IR::ForInStatement *s) {
         builder.supressStatementSemi();
         visit(s->decl, "decl");
     } else {
-        visit(s->ref, "ref");
+        auto *decl = resolveUnique(s->ref->path->name, ResolutionType::Any);
+        if (auto *di = decl->to<IR::Declaration_Variable>()) {
+            builder.supressStatementSemi();
+            visit(di, "decl");
+        } else {
+            visit(s->ref, "ref");
```

Comment:
Can this branch be used for a code that is accepted by the parser?

Reply:
No -- any code that hits this branch will dump in a form that can't be parsed.  Perhaps this should be a BUG instead, as this should not happen -- there should always be a declaration of the symbol in scope.